### PR TITLE
support for tuflowfv

### PR DESCRIPTION
**Pull request by PeterPetrik**

### CODE REVIEW

**Comment by PeterPetrik** [comment]
fixes #171 
fixes #154

File: mdal/frmts/mdal_3di.cpp [hunk 1]
```diff
@@ -121,10 +121,10 @@ void MDAL::Driver3Di::addBedElevation( MemoryMesh *mesh )
   size_t faceCount = mesh->facesCount();
 
   // read Z coordinate of 3di computation nodes centers
-  int ncidZ = mNcFile.getVarId( "Mesh2DFace_zcc" );
-  double fillZ = mNcFile.getFillValue( ncidZ );
+  int ncidZ = mNcFile->getVarId( "Mesh2DFace_zcc" );
+  double fillZ = mNcFile->getFillValue( ncidZ );
   std::vector<double> coordZ( faceCount );
-  if ( nc_get_var_double( mNcFile.handle(), ncidZ, coordZ.data() ) )
+  if ( nc_get_var_double( mNcFile->handle(), ncidZ, coordZ.data() ) )
```
Comment (MarcusUrban) [inline mdal/frmts/mdal_3di.cpp]: No Exception throw for this fail is needed?

File: mdal/frmts/mdal_ascii_dat.cpp [hunk 2]
```diff
@@ -127,16 +127,17 @@ void MDAL::DriverAsciiDat::loadOldFormat( std::ifstream &in,
   while ( std::getline( in, line ) );
 
   if ( !group || group->datasets.size() == 0 )
-    EXIT_WITH_ERROR( MDAL_Status::Err_UnknownFormat );
+    EXIT_WITH_ERROR( MDAL_Status::Err_UnknownFormat )
 
-  group->setStatistics( MDAL::calculateStatistics( group ) );
+    group->setStatistics( MDAL::calculateStatistics( group ) );
```
Comment (MarcusUrban) [inline mdal/frmts/mdal_ascii_dat.cpp]: Extra tab

File: mdal/frmts/mdal_ascii_dat.cpp [hunk 3]
```diff
@@ -106,8 +106,8 @@ void MDAL::DriverAsciiDat::loadOldFormat( std::ifstream &in,
       size_t fileNodeCount = toSizeT( items[1] );
       size_t meshIdCount = maximumId( mesh ) + 1;
       if ( meshIdCount != fileNodeCount )
-        EXIT_WITH_ERROR( MDAL_Status::Err_IncompatibleMesh );
```
Comment (MarcusUrban) [inline mdal/frmts/mdal_ascii_dat.cpp]: +1

File: mdal/frmts/mdal_3di.cpp (hunk 1 above)
Comment (PeterPetrik) [reply mdal/frmts/mdal_3di.cpp]: probably not

**Comment by saberraz** [comment]
I cannot build QGIS master with MDAL master:
```
[310/315] Building CXX object src/providers/mdal/CMakeFiles/mdalprovider.dir/qgsmdalprovider.cpp.o
FAILED: src/providers/mdal/CMakeFiles/mdalprovider.dir/qgsmdalprovider.cpp.o 
ccache /usr/bin/c++  -DPROJ_VERSION_MAJOR=6 -DQT_CONCURRENT_LIB -DQT_CORE_LIB -DQT_DBUS_LIB -DQT_DISABLE_DEPRECATED_BEFORE=0 -DQT_GUI_LIB -DQT_NETWORK_LIB -DQT_NO_CAST_TO_ASCII -DQT_POSITIONING_LIB -DQT_PRINTSUPPORT_LIB -DQT_QML_LIB -DQT_QUICKWIDGETS_LIB -DQT_QUICK_LIB -DQT_SERIALPORT_LIB -DQT_SQL_LIB -DQT_SVG_LIB -DQT_UITOOLS_LIB -DQT_WEBKITWIDGETS_LIB -DQT_WEBKIT_LIB -DQT_WIDGETS_LIB -DQT_XML_LIB -DSIP_VERSION=0x041312 -DWITH_BINDINGS -DWITH_QTWEBKIT -Dmdalprovider_EXPORTS -Isrc/providers/mdal/mdalprovider_autogen/include -I. -I../src/core -I../src/core/mesh -I../src/core/expression -I../src/core/geometry -I../src/core/metadata -I../src/core/symbology -I../src/gui -I../src/gui/auth -I../external -I../external/nlohmann -Isrc/core -Isrc/gui -Isrc/ui -I/usr/include/hdf5/serial -I/usr/include/libxml2 -isystem /usr/local/include -isystem /usr/include/x86_64-linux-gnu/qt5 -isystem /usr/include/x86_64-linux-gnu/qt5/QtXml -isystem /usr/include/x86_64-linux-gnu/qt5/QtCore -isystem /usr/lib/x86_64-linux-gnu/qt5/mkspecs/linux-g++ -isystem /usr/include/x86_64-linux-gnu/qt5/QtGui -isystem /usr/include/x86_64-linux-gnu/qt5/QtWidgets -isystem /usr/include/x86_64-linux-gnu/qt5/QtSvg -isystem /usr/include/x86_64-linux-gnu/qt5/QtPrintSupport -isystem /usr/include/x86_64-linux-gnu/qt5/QtNetwork -isystem /usr/include/x86_64-linux-gnu/qt5/QtWebKitWidgets -isystem /usr/include/x86_64-linux-gnu/qt5/QtWebKit -isystem /usr/include/x86_64-linux-gnu/qt5/QtSql -isystem /usr/include/x86_64-linux-gnu/qt5/QtConcurrent -isystem /usr/include/x86_64-linux-gnu/qt5/QtSerialPort -isystem /usr/include/x86_64-linux-gnu/qt5/QtPositioning -isystem /usr/include/x86_64-linux-gnu/qt5/QtDBus -isystem /usr/include/x86_64-linux-gnu/qt5/QtUiTools -isystem /usr/include/x86_64-linux-gnu/qt5/QtQuickWidgets -isystem /usr/include/x86_64-linux-gnu/qt5/QtQuick -isystem /usr/include/x86_64-linux-gnu/qt5/QtQml -Wall -Wextra -Wno-long-long -Wformat-security -Wno-strict-aliasing -Wno-redundant-move -Wno-deprecated-copy -g -fPIC -fvisibility=hidden   -fPIC -std=gnu++11 -MD -MT src/providers/mdal/CMakeFiles/mdalprovider.dir/qgsmdalprovider.cpp.o -MF src/providers/mdal/CMakeFiles/mdalprovider.dir/qgsmdalprovider.cpp.o.d -o src/providers/mdal/CMakeFiles/mdalprovider.dir/qgsmdalprovider.cpp.o -c ../src/providers/mdal/qgsmdalprovider.cpp
../src/providers/mdal/qgsmdalprovider.cpp: In member function ‘virtual bool QgsMdalProvider::persistDatasetGroup(const QString&, const QgsMeshDatasetGroupMetadata&, const QVector<QgsMeshDataBlock>&, const QVector<QgsMeshDataBlock>&, const QVector<double>&)’:
../src/providers/mdal/qgsmdalprovider.cpp:201:39: error: cannot convert ‘bool’ to ‘MDAL_DataLocation’
  201 |                       meta.dataType() == QgsMeshDatasetGroupMetadata::DataOnVertices,
      |                       ~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
      |                                       |
      |                                       bool
In file included from ../src/providers/mdal/qgsmdalprovider.h:23,
                 from ../src/providers/mdal/qgsmdalprovider.cpp:20:
/usr/local/include/mdal.h:184:23: note:   initializing argument 3 of ‘void* MDAL_M_addDatasetGroup(MeshH, const char*, MDAL_DataLocation, bool, DriverH, const char*)’
  184 |     MDAL_DataLocation dataLocation,
      |     ~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~
../src/providers/mdal/qgsmdalprovider.cpp: In member function ‘virtual QgsMeshDatasetGroupMetadata QgsMdalProvider::datasetGroupMetadata(int) const’:
../src/providers/mdal/qgsmdalprovider.cpp:431:23: error: ‘MDAL_G_isOnVertices’ was not declared in this scope; did you mean ‘isOnVertices’?
  431 |   bool isOnVertices = MDAL_G_isOnVertices( group );
      |                       ^~~~~~~~~~~~~~~~~~~
      |                       isOnVertices

```

**Comment by saberraz** [comment]
Obviously, me being dumb!

It builds with no error (but some warning) with this PR: https://github.com/qgis/QGIS/pull/33153

**Pull request merged**